hack/arktos-worker-up.sh

```diff
@@ -17,29 +17,56 @@
 set -e
 die() { echo "$*" 1>&2 ; exit 1; }
 
-if [[ ! -n "${KUBELET_IP}" ]]; then
-    die "KUBELET_IP env var not set"
-fi
-
 # install cni plugin related packages
 KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source ${KUBE_ROOT}/hack/lib/common-var-init.sh
 source ${KUBE_ROOT}/hack/arktos-cni.rc
 
-mkdir -p /tmp/arktos
+# creates a self-contained kubeconfig: args are sudo, dest-dir, ca file, host, port, client id, token(optional)
+function write_client_kubeconfig {
+    local sudo=$1
+    local dest_dir=$2
+    local ca_file=$3
+    local api_host=$4
+    local api_port=$5
+    local client_id=$6
+    local token=${7:-}
+    local protocol=${8:-"https"}
```

Comment:
is the prtocol default https or http?

Reply:
You are right, we don't support https now. It should default to http.

> is the prtocol default https or http?